### ISSUE TITLE
Debug Adapter Protocol: Is there any way to send an event to vscode to initiate a variable request

### ISSUE THREAD

**Issue by maziac** [Microsoft/vscode#49008]
I'm writing a debug adapter for a Z80 emulator. I would like to change some variables (e.g. memory or register values) at the emulator and then display the result in vscode. Currently I have put the registers in the VARIABLES area and on every step-over (next) vscode sends a variable request to get the variables updated. That works fine.

But how should it work if e.g. the debugger is paused and then a register value is changed manually at the emulator.

I would like to inform vscode about the new value but there doesn't seem to be any API for this.

Every update request is initated from vscode as it seems. Or am I missing something?

- VSCode Version: 1.22.2
- OS Version: OSX 10.13.4

**Comment by isidorn** [comment]
@maziac you are correct. There currently is no way to do this. You would need something new like a `VariableChangeEvent`.
Forwarding this to @weinand to consider for the future
